### PR TITLE
Simplify development setup

### PR DESCRIPTION
**Pull request by amazon-meaisiah**
*Issue #, if available:*

*Description of changes:*

- The bulk of the meaningful changes are in [this commit](https://github.com/awslabs/aws-api-gateway-developer-portal/commit/e1abb50812d100f0788b636a1be5d6f2a41c010e).
	- I recasted the scripts to automate most of the manual stuff away. In particular, installing all the dependencies and getting your dev environment set up is a single simple, easy command.
	- I revised the setup to make it a much more fluid process to get started.
	- I hand-rolled a task runner to reduce a lot of the otherwise necessary shell magic.
	- I removed dependency on OS-dependent utilities like `cp` and `rm` at build time. (I also killed a script's unnecessary dependency on zsh.)
	- I moved all the scripts from `dev-portal` to the root so it's easier to work with in the general case.
	- I now revert the generated static assets to the currently checked-out versions after deploying as part of the "release" task, to avoid accidentally committing regenerated assets in that common case. (You can still manually build and deploy without resetting, just not through `node run release`.)
- The rest of the commits are atomic and explain it very well.

Note: this has no impact on the developer portal as deployed through this repo. It's all just DX improvements.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by amazon-meaisiah** [comment]
Here's the real diff for this PR: https://github.com/awslabs/aws-api-gateway-developer-portal/compare/dev-tweaks..7b0b1d7830f61377dcf7e1e920fc3db288f14668

File: scripts/npm.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  '/lambdas/backend',
+  '/lambdas/catalog-updater',
+  '/lambdas/listener',
+  '/lambdas/static-asset-uploader'
+]
+
+async function execute (args) {
+  for (const folder of folders) {
+    const resolved = path.join(__dirname, '..', folder.slice(1))
+    console.error(`\x1b[32mExecuting in folder:\x1b[0m \x1b[34m${folder}\x1b[0m`)
```
Comment (etsung-aws) [inline scripts/npm.js]: does this console error formatting work on all platforms?

File: scripts/npm.js [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
-}
+const { run, packageList } = require('./internal/util')
 
-visit('')
-  .then(() => visit('dev-portal'))
-  .then(() => visit('lambdas/backend'))
-  .then(() => visit('lambdas/catalog-updater'))
-  .then(() => visit('lambdas/listener'))
-  .then(() => visit('lambdas/static-asset-uploader'))
+;(async () => {
```
Comment (etsung-aws) [inline scripts/npm.js]: why iife?

Comment (amazon-meaisiah) [reply scripts/npm.js]: Look for the `await`. Easier than writing a `reduce` + `.then` loop.

**Pull request merged**